code/to_bsdd.py

```diff
@@ -57,18 +56,45 @@ def skip_by_package(element):
     
 HTML_TAG_PATTERN = re.compile('<.*?>')
 MULTIPLE_SPACE_PATTERN = re.compile(r'\s+')
-CHANGE_LOG_PATTERN = re.compile(r'\{\s*\.change\-\w+\s*\}.+', flags=re.DOTALL)
+# CHANGE_LOG_PATTERN = re.compile(r'\{\s*\.change\-\w+\s*\}.+', flags=re.DOTALL)
+CURLY_BRACKET_PATTERN = re.compile('{.*?}')
+
 def strip_html(s):
+    try:
+        s1 = html.unescape(s)
+    except TypeError:
+        # error when name contains link, for example: "https://github.com/buildingSMART/IFC4.3.x-development/edit/master/docs/schemas/core/IfcProductExtension/Types/IfcAlignmentTypeEnum.md#L0 has no content"
+        s1 = ''
+    i = s1.find('\\n')
+    if i != -1:
+        s1 = s1[:i]
+    s2 = re.sub(HTML_TAG_PATTERN, ' ', s1)
+    s3 = re.sub(CURLY_BRACKET_PATTERN, ' ', s2)
+    s4 = re.sub('   ', ' ', s3)
+    s5 = re.sub('  ', ' ', s4)
+    return s5
+
+def split_at_word(w, s):
+    if w in s.lower():
+        if not w+"ing" in s.lower():
+            s = re.sub(w, " "+w+" ", s.lower())
+    if isinstance(s, list):
+        s = " ".join(s)
     return s
-    S = html.unescape(s or '')
-    i = S.find('\n')
-    return re.sub(HTML_TAG_PATTERN, '', S)
-    
+
+def split_words(s):
+    # hard-coded list of words found in enumerations to split the ALLCAPS phrases into indivudual words. List might not be complete
+    for w in ['current','switch','order','recovery','loading','barrier','reel','basin','fountain','packaged','nozzle','tube','plant','injection','assisted','element','vertical','turbine','heating','disassembly','button','exchange','component','security','deflector','chair','cabinet','assembly','actuator','meter','sensor','object','detection','station','depth','controller','terminal','center','frame','electric','exchangers','outlet','server','plate','expansion','exhaust','damper','shell','pump','filter','conveyor','rail','solar','surcharge','excavation','combustion','draft','mechanical','constant','coil','cooled','cooler','evaporative','pavement','top','column','traffic','crossing','side','road','vehicle','island','gear','super','event','adiabatic','segment','marker','structure','ground','channel','pressure','shift','prevention','tray','provision','soil','preloaded','water','cold','hot','cable','domestic','power','generation','solid','waste','unit','carrier','duct','protection','disconnector','surfacing','breaker','wall','flow','curve','limiter','board','chamber','panel','acoustic','fire','inspection','work','marking','transverse','rumble','strip','surface','maintenance','of','system','fixed','transmission','network','machine','device','equipment','sound','stud','connector','section','inventory','bill','quantities','compacted','drained','tower','indirect','direct','media','rigid','random','gravity','relief','air']:
```

Comment:
Did you consider sth like `from nltk.corpus import words`? I think you also need to sort based on length, try candidates first.

Reply:
I considered it, but it has too many bizarre or short words or even individual letters. I think such a manually curated list makes more sense for IFC. We only need it once. Probably some words are still missing, but we can always add when we find them. Still, better to leave them joined, than split too much.